### PR TITLE
remove convert from CreatorLockForm

### LINKED ISSUE #1068: refactor CreatorLockForm tests to not need the convert prop to be false
Currently, the `CreatorLockForm` component has a `convert` prop, which is used to tell the constructor not to convert `keyPrice` and `expirationDuration` to form-friendly values from lock values.

In the app, `keyPrice` is represented in wei, and in the form, it is represented in eth.
In the app, `expirationDuration` is represented in seconds, and in the form, it is represented in days.

The unit tests for `CreatorLockForm` test several error conditions that cannot be tested simply by using `react-testing-library`'s `fireEvent.change` event trigger, because jsdom's implementation of the number input will not allow anything to be fired on it that is not a number. This is the behavior of opera, but unfortunately, both chrome and firefox allow users to input non-number characters.

To remove the convert prop, we will need to find another way to test the form field validation edge cases, and that is the scope of this issue.

ref: #1064

### PR DESCRIPTION
**Pull request by cellog**
# Description

Fixes #1068 by removing convert. In order to make it possible to test against invalid values, we are extremely permissive on input. So, when invalid input is encountered, it is simply left untouched. This way, we leave the error conditions to the form itself to display. In other words, if passed a cockamamie lock, the edit form will simply refuse to allow you to save it until the errors are fixed. Screenshots to show I didn't break anything :)

screen shot 2019-01-23 at 11 50 39 pm
screen shot 2019-01-23 at 11 50 28 pm
screen shot 2019-01-23 at 11 49 44 pm
screen shot 2019-01-23 at 11 49 37 pm
screen shot 2019-01-23 at 11 50 54 pm

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [X] My code follows the style guidelines of this project, including naming conventions
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [X] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

PS: [Read how to write the perfect pull request](https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/)

### CODE REVIEW

File: unlock-app/src/__tests__/components/creator/CreatorLockForm.test.js [hunk 1]
```diff
@@ -99,9 +93,9 @@ describe('CreatorLockForm', () => {
       }
     })
     it('key expiration is a negative number', () => {
-      const wrapper = makeLockForm({ expirationDuration: -1 })
+      const wrapper = makeLockForm({ expirationDuration: -2 * 86400 })
```
Comment (julien51) [inline unlock-app/src/__tests__/components/creator/CreatorLockForm.test.js]: Nit: when using numeric values like this, feel free to put the corresponding "human" format like "2 days" as comments. This makes it easier to skip the math ;)

File: unlock-app/src/components/creator/CreatorLockForm.js [hunk 2]
```diff
@@ -37,15 +37,31 @@ export class CreatorLockForm extends React.Component {
...
     let maxNumberOfKeys = props.maxNumberOfKeys
-    if (props.convert) {
-      keyPrice = Web3Utils.fromWei(keyPrice, props.keyPriceCurrency)
-      expirationDuration = expirationDuration / props.expirationDurationUnit
-      // unlimited keys is represented differently in the frontend and the backend,
-      // so when a lock has `maxNumberOfKeys` set to UNLIMITED_KEYS_COUNT,
-      // convert it to infinity symbol for frontend display
-      if (maxNumberOfKeys === UNLIMITED_KEYS_COUNT) {
-        maxNumberOfKeys = INFINITY
+    try {
```
Comment (julien51) [inline unlock-app/src/components/creator/CreatorLockForm.js]: Please add a comment to explain the rationale for the try catch thing. IIf I understanding things correctly this is because we invoke this with different formats which is probably something we should fix  (I k now I have an issue assigned to me for that!)

File: unlock-app/src/__tests__/components/creator/CreatorLockForm.test.js (hunk 1 above)
Comment (cnasc) [reply unlock-app/src/__tests__/components/creator/CreatorLockForm.test.js]: Just for the sake of argument, when I do things like this I like to have a named variable like `secondsInADay` so that you end up with `2 * secondsInADay` which is a bit more self-documenting and resilient against change (e.g., if we change it in the future to being 3 days, there won't be a comment that can be out of sync with the code).

Comment (cellog) [reply unlock-app/src/__tests__/components/creator/CreatorLockForm.test.js]: I like that @cnasc thanks

File: unlock-app/src/components/creator/CreatorLockForm.js (hunk 2 above)
Comment (cellog) [reply unlock-app/src/components/creator/CreatorLockForm.js]: 👍
